### PR TITLE
[CDAP-18242] Check for CREATE Permission on Compute Profile Creation

### PR DESCRIPTION
**Pull request by dli357**
See [CDAP-18242](https://cdap.atlassian.net/browse/CDAP-18242) for details.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java [hunk 1]
```diff
@@ -181,7 +193,19 @@ public void writeProfile(FullHttpRequest request, HttpResponder responder,
                            @PathParam("namespace-id") String namespaceId,
                            @PathParam("profile-name") String profileName) throws Exception {
     ProfileId profileId = getValidatedProfile(namespaceId, profileName);
-    accessEnforcer.enforce(profileId, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+    boolean profileExists;
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java]: The code are almost identical? Can you refactor into a private method?

**Review by chtyim — COMMENTED** [review]
Do we have any permission related unit-test for this?

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java [hunk 2]
```diff
@@ -105,7 +105,19 @@ public void getSystemProfile(HttpRequest request, HttpResponder responder,
...
+    try {
+      profileService.getProfile(profileId);
+      profileExists = true;
+    } catch (NotFoundException e) {
+      profileExists = false;
+    }
+    if (profileExists) {
+      accessEnforcer.enforce(profileId, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+    } else {
+      accessEnforcer.enforceOnParent(EntityType.PROFILE, profileId.getParent(), authenticationContext.getPrincipal(),
```
Comment (tivv) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java]: As soon as you know the profile name you can check it with `enforce`, not `enforceOnParent`

Comment (dli357) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java]: Fixed

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java (hunk 1 above)
Comment (dli357) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java]: Fixed

**Review by tivv — APPROVED** [review]

**Review by dli357 — COMMENTED** [review]
@chtyim Added unit tests, thanks!

**Pull request merged**